resctrl/utils.go

```diff
@@ -0,0 +1,285 @@
+		if err != nil {
+			return "", err
+		}
+		for _, thread := range processThreads {
+			err = intelrdt.WriteIntelRdtTasks(monGroupPath, thread)
+			if err != nil {
+				secondError := os.Remove(monGroupPath)
+				if secondError != nil {
+					return "", fmt.Errorf("%v : %v", err, secondError)
+				}
+				return "", err
+			}
+		}
+	}
+
+	return monGroupPath, nil
+}
+
+func getPids(containerName string) ([]int, error) {
+	if containerName == "" {
+		return nil, fmt.Errorf(noContainerError)
+	}
+	pids, err := cgroups.GetAllPids(filepath.Join(pidsPath, containerName))
+	if err != nil {
+		return nil, fmt.Errorf("couldn't obtain pids: %v", err)
+	}
+	return pids, nil
+}
+
+func getAllProcessThreads(threadFiles []os.FileInfo) ([]int, error) {
+	processThreads := make([]int, 0)
+	for _, file := range threadFiles {
+		pid, err := strconv.Atoi(file.Name())
+		if err != nil {
+			return nil, fmt.Errorf("couldn't parse %q file: %v", file.Name(), err)
+		}
+		processThreads = append(processThreads, pid)
+	}
+
+	return processThreads, nil
+}
+
+func findControlGroup(pids []string) (string, error) {
+	if len(pids) == 0 {
+		return "", fmt.Errorf(noPidsError)
+	}
+	availablePaths, err := filepath.Glob(filepath.Join(rootResctrl, "*"))
+	if err != nil {
+		return "", err
+	}
+
+	for _, path := range availablePaths {
+		switch path {
+		case filepath.Join(rootResctrl, cpusFileName):
+			continue
+		case filepath.Join(rootResctrl, cpusListFileName):
+			continue
+		case filepath.Join(rootResctrl, infoDirName):
+			continue
+		case filepath.Join(rootResctrl, monDataDirName):
+			continue
+		case filepath.Join(rootResctrl, monGroupsDirName):
+			continue
+		case filepath.Join(rootResctrl, schemataFileName):
+			continue
+		case filepath.Join(rootResctrl, tasksFileName):
+			inGroup, err := checkControlGroup(rootResctrl, pids)
+			if err != nil {
+				return "", err
+			}
+			if inGroup {
+				return rootResctrl, nil
+			}
+		default:
+			inGroup, err := checkControlGroup(path, pids)
+			if err != nil {
+				return "", err
+			}
+			if inGroup {
+				return path, nil
+			}
+		}
+	}
+
+	return "", fmt.Errorf("there is no available control group")
+}
+
+func checkControlGroup(path string, pids []string) (bool, error) {
+	if len(pids) == 0 {
+		return false, fmt.Errorf(noPidsError)
+	}
+
+	taskFile, err := fscommon.ReadFile(path, "tasks")
+	if err != nil {
+		return false, fmt.Errorf("couldn't obtain pids: %v", err)
+	}
+
+	for _, pid := range pids {
+		if !strings.Contains(taskFile, pid) {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func readStat(path string) (uint64, error) {
+	context, err := ioutil.ReadFile(path)
+	if err != nil {
+		return 0, err
+	}
+
+	stat, err := strconv.ParseUint(string(bytes.TrimSpace(context)), 10, 64)
+	if err != nil {
+		return stat, fmt.Errorf("unable to parse %q as a uint from file %q", string(context), path)
+	}
+
+	return stat, nil
+}
+
+func getStats(path string) (intelrdt.Stats, error) {
```

Comment:
s/getStats/getStatistics/

Reply:
`UpdateStats()` uses `getStats()`.
Why do you think that I cannot use short form also here?

Reply:
Fair enough.